packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -69,80 +77,95 @@ def create_client(
     measurement_service_class: str,
     module_name: Optional[str],
     class_name: Optional[str],
+    all: Optional[bool],
     directory_out: Optional[str],
 ) -> None:
     """Generates a Python Measurement Plug-In Client module for the measurement service.
 
     You can use the generated module to interact with the corresponding measurement service.
 
     MEASUREMENT_SERVICE_CLASS: The service class of the measurement.
+    Provide comma separated service class to generate multiple clients.
     """
     channel_pool = GrpcChannelPool()
     discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
     built_in_import_modules: List[str] = []
     custom_import_modules: List[str] = []
 
-    if module_name is None or class_name is None:
-        base_service_class = measurement_service_class.split(".")[-1]
-        base_service_class = remove_suffix(base_service_class)
-        if not base_service_class.isidentifier():
+    if all:
+        service_class_list = get_service_class_list(discovery_client)
+    else:
+        if not measurement_service_class:
+            raise click.ClickException("Measurement service class cannot be empty.")
+        service_class_list = measurement_service_class.split(",")
```

Comment:
Instead of getting the service class names as a comma separated list, can we pass them as separate parameters?

Reply:
May be [nargs](https://click.palletsprojects.com/en/8.1.x/arguments/#variadic-arguments) might help here.

Reply:
Sure, I have updated it with `nargs=-1`.